x-pack/elastic-agent/docs/elastic-agent-dynamic-inputs.asciidoc

```diff
@@ -0,0 +1,837 @@
+[[dynamic-input-configuration]]
+[role="xpack"]
+= Dynamic input configuration
+
+//TODO: Break into multiple source files to make this info easier to change over
+//time. Maybe move to a separate directory.
+
+//PH: I'm not sure what the best heading for this topic is. What are users
+//likely to search on?
+
+The inputs defined in {agent} support variable replacement and conditions. When
+used together, these capabilities offer a lot of flexibility for you to define
+configurations based on information available from the running environment and
+decide when to run a defined input.
+
+//PH: Do we want to say something about container monitoring here, or is
+//it too early? 
+
+Let's consider a unique agent policy that is deployed on two machines: a Linux
+machine named "linux-app" and a Windows machine named "winapp":
+
+[source,yaml]
+----
+inputs:
+ - type: logfile
+   streams:
+    - paths: /var/log/${host.name}/another.log
+      condition: ${host.platform} == "linux"
```

Comment:
Is the purpose of this section to start with an overview? It includes conditions and variable substitution, but we haven't defined what those are yet.

Reply:
Yah I'll work a little on the intro a little more, but probably won't have time to polish it a lot before 7.10.